source/MainWindow.cpp

```diff
@@ -1393,29 +1392,69 @@ MainWindow::_BuildEncodeProgress()
 
 
 void
-MainWindow::_BuildLine() // ask all the views what they hold, reset the command string
+MainWindow::_BuildLine() // update the ffmpeg commandline
 {
-	BString source_filename(fSourceTextControl->Text());
-	BString output_filename(fOutputTextControl->Text());
-	source_filename.Trim();
-	output_filename.Trim();
-	fCommand = kFFMpeg;
-	fCommand << " -i \"" << source_filename << "\""; // append the input file name
+	// split existing commandline into tokens
+	fCommand = fCommandlineTextControl->Text();
+	fCommand.Trim();
+	BStringList tokens_raw;
+	fCommand.Split(" ", true, tokens_raw);
+
+	// consolidate parts of quoted strings into single tokens
+	fCommandLineTokens.MakeEmpty();
+	for (int32 token_idx=0; token_idx<tokens_raw.CountStrings(); ++token_idx) {
+		BString token = tokens_raw.StringAt(token_idx);
+		if (token.StartsWith("\""))
+			if (!token.EndsWith("\""))
```

Comment:
I think we do need curly brackets here, because the if-block stretches over several lines.